### PR TITLE
[WIP] OSD-10062 Add ManagedNotification API

### PR DESCRIPTION
**Pull request by mrbarge**
### What type of PR is this?

Feature

### What this PR does / why we need it?

This adds a `ManagedNotification` API and associated generated code, plus additional helper functions and associated test cases.

This also adds additional permissions to the `ocm-agent` role to be able to read and update `ManagedNotification` resources.

This PR is still pending review of the CRD design per [OSD-9787](https://issues.redhat.com/browse/OSD-9787), but needed to be done to really validate the design, and might give people a more useful insight into how the CRD will look/feel.

### Which Jira/Github issue(s) this PR fixes?

[OSD-9787](https://issues.redhat.com/browse/OSD-9787)
[OSD-10062](https://issues.redhat.com/browse/OSD-10062)

### Pre-checks (if applicable):
- [X] Tested latest changes against a cluster
- [X] Ran `make generate` command locally to validate code changes
- [X] Included documentation changes with PR

### CODE REVIEW

**Comment by mrbarge** [comment]
This is a work in progress whilst design is reviewed; it may change.
/hold

File: deploy/20_ocm-agent.Role.yaml [hunk 1]
```diff
@@ -19,3 +19,13 @@ rules:
       - patch
       - update
       - watch
+  - apiGroups:
+      - ocmagent.managed.openshift.io
+    resources:
+      - managednotifications
+    verbs:
```
Comment (bmeng) [inline deploy/20_ocm-agent.Role.yaml:26]: should we give all the permissions to the role since the CR is under the ocmagent api group?

Comment (mrbarge) [reply deploy/20_ocm-agent.Role.yaml:26]: I couldn't really see any reason why the `ocm-agent` SA would need to delete the ManagedNotification CR.

**Pull request closed without merging**